File: frontend/packages/shared/src/types/api/DeploymentsResponse.ts
import type { KubernetesDeployment } from './KubernetesDeployment';
import type { PipelineDeployment } from './PipelineDeployment';

export interface DeploymentsResponse {
  pipelineDeploymentList: PipelineDeployment[];
  kubernetesDeploymentList: KubernetesDeployment[];
}

File: backend/src/Designer/Models/DeploymentsResponse.cs
using System.Collections.Generic;
using Altinn.Studio.Designer.Repository.Models;
using Altinn.Studio.Designer.TypedHttpClients.KubernetesWrapper;

namespace Altinn.Studio.Designer.Models
{
    /// <summary>
    /// Deployments response
    /// </summary>
    public class DeploymentsResponse
    {
        public List<DeploymentEntity> PipelineDeploymentList { get; set; }

        public List<KubernetesDeployment> KubernetesDeploymentList { get; set; }
    }
}
